source/server/server.cc

```diff
@@ -379,6 +397,24 @@ void InstanceImpl::initialize(Network::Address::InstanceConstSharedPtr local_add
                                     messageValidationContext().staticValidationVisitor(), *api_);
   bootstrap_config_update_time_ = time_source_.systemTime();
 
+#ifdef ENVOY_PERFETTO
+  perfetto::TracingInitArgs args;
+  // Include in-process events only.
+  args.backends = perfetto::kInProcessBackend;
+  perfetto::Tracing::Initialize(args);
+  perfetto::TrackEvent::Register();
+  perfetto::TraceConfig cfg;
+  // TODO(rojkov): make the tracer configurable with either `Perfetto`'s native
+  // message or custom one embedded into Bootstrap.
+  cfg.add_buffers()->set_size_kb(1024);
```

Comment:
Since tracers only seem to be written at server destruction, I'm a bit confused:
1. What is the meaning of this buffer vs. presumably the indefinite trace accumulation in memory?
2. What if a user wants to capture snapshots of performance traces for a live server?

Reply:
By default it's a ring buffer that needs to be flushed to a file (or another sink) before it gets overwritten. I've modified the behavior to write to the file continuously in the tracing thread. Now the file gets just flushed and closed at server destruction.

Currently switching tracing on and off for a live server is not implemented as I use this SDK for development only. But this functionality could be added through the admin interface.